### PR TITLE
Update benchmark tests

### PR DESCRIPTION
**Pull request by abidwael**
- Removes `pytest_benchmarks.yaml` and includes the quality regression benchmarks as part of the main `pytest.yaml` workflow. (they take an additional 1.5-2min)
- Improvements to the `ludwig.benchmarking` module for better error surfacing.

### CODE REVIEW

File: .github/workflows/pytest_benchmarks.yml [hunk 1]
```diff
@@ -3,9 +3,14 @@
 
 name: benchmark
 
+#on:
+#  push:
+#    branches: ["master", "release-*"]
 on:
   push:
     branches: ["master", "release-*"]
+  pull_request:
```
Comment (justinxzhao) [inline .github/workflows/pytest_benchmarks.yml:12]: Now that you've verified that the benchmark tests work with your change, is the plan to revert these files before merging the PR?

Comment (abidwael) [reply .github/workflows/pytest_benchmarks.yml]: The plan is to add them to the main test suite (`pytest.yml`) and remove `pytest_benchmarks.yml`

File: tests/regression_tests/benchmark/configs/sarcos.gbm.yaml [hunk 2]
```diff
@@ -107,5 +107,3 @@ trainer:
   num_boost_round: 382
   num_leaves: 434
 model_type: gbm
-backend:
-  type: local
```
Comment (arnavgarg1) [inline tests/regression_tests/benchmark/configs/sarcos.gbm.yaml:111]: Is there a reason we're removing backend as local? Should we potentially consider setting it to Ray?

**Review by tgaddair — APPROVED** [review]

**Review by justinxzhao — APPROVED** [review]

File: tests/regression_tests/benchmark/configs/sarcos.gbm.yaml (hunk 2 above)
Comment (abidwael) [reply tests/regression_tests/benchmark/configs/sarcos.gbm.yaml:111]: I believe default is ray

**Pull request merged**